### ISSUE TITLE
Licensing

### ISSUE THREAD

**Issue by paragonie-scott** [jedisct1/libsodium-php#127]
The [pull request against php-src](https://github.com/php/php-src/pull/2560) is being blocked by a licensing discrepancy.

This repository is released under a BSD license. PHP requires the PHP license (which is, itself, a variant of a BSD license).

The contributors to this repository are:

- [x] @jedisct1 
- [x] @paragonie-scott
- [x] @remicollet (PHP internals developer)
- [x] @nikic (PHP internals developer)
- [x] @xor-zz 
- [x] @leonklingele 
- [x] @fredemmott 
- [x] @glensc 
- [x] @kherge 
- [x] @cmckni3 
- [x] @mc10 
- [x] @wlejon 
- [x] @Jan-E 

If we can get everyone who has contributed to this PHP extension to OK changing the license to get libsodium into core, then we can make it happen without drama or issue. I'm not sure what the path looks like without a unanimous consent. It might be easy since the licenses are so similar.

CCing @remicollet and @sgolemon who are the PHP 7.2 release managers.

**Comment by paragonie-scott** [comment]
I'm okay with what I've contributed being relicensed for the purpose of getting libsodium into the PHP core.

**Comment by leonklingele** [comment]
Great to see this getting merged into stdlib. Land it!

**Comment by merlingore** [comment]
Would love to see this in core! Great job all.

**Comment by cmckni3** [comment]
Sounds good to me. 👍

**Comment by wlejon** [comment]
Yes, please.

**Comment by paragonie-scott** [comment]
Just to be clear: This will affect the code in the php-src pull request, but it doesn't need to necessarily apply to this github repository.

**Comment by jedisct1** [comment]
It doesn't apply to this repository. Since this is a more restrictive license, PHP can use code from this repository, but the other way round is impossible.

Dual licensing a PHP extension is apparently an impossible thing. Sad.

But we don't really have the choice if we want this to be part of the PHP tarballs.

**Comment by fredemmott** [comment]
The changes I made are the property of Facebook, and may be used under the same licenses as PHP itself.

**Comment by kherge** [comment]
Sounds great to me! :+1:

**Comment by nikic** [comment]
I'm fine with whatever needs to be done.

> Dual licensing a PHP extension is apparently an impossible thing. Sad.

@sgolemon Please comment on this. I think there might have been a miscommunication at some point. Please clarify whether this was the intended meaning of [this comment](https://github.com/php/php-src/pull/2560#issuecomment-312274755) (I have interpreted it this way as well, but I'm not sure if it was really intended) and explain in more detail why dual licensing is not possible.

PS: I might be somewhat pissed about how this issue has been handled from the PHP side at this point.

**Comment by mc10** [comment]
I'm fine with the license change.

**Comment by glensc** [comment]
i'm ok as well with license change

**Comment by sgolemon** [comment]
How do either of you interpret "If that's a blocker for you, we can potentially take the BSD license as-is" as being meaning that dual-licensing or any other option is somehow off the table?

I don't know how I can be more clear.

PS: I absolutely am pissed about how this issue has been NOT handled at all from the sodium side.

**Comment by Jan-E** [comment]
I am OK with whatever needs to be done.

**Comment by remicollet** [comment]
I'm fine with the license change to PHP.

**Comment by remicollet** [comment]
Ping @xor-zz (only missing answer)

**Comment by kelunik** [comment]
What are the options if @xor-zz doesn't respond? There are [5 commits](https://github.com/jedisct1/libsodium-php/commits?author=xor-zz) with 3 being simply typos or one line changes. One is a test, that could be removed. Another one is https://github.com/jedisct1/libsodium-php/commit/dd228c1463f80afa4e801c0bd236aed0169c8b69 which I'm not sure could be written substantially differently, so it might not be copyrightable, but totally not sure on that.

**Comment by xor-zz** [comment]
Sure. Go for it.

**Comment by padraigdoran** [comment]
It's best to make the needed licence changes to be included in the standard PHP library.

**Comment by paragonie-scott** [comment]
Thanks everyone. I'm closing this and shifting gears to https://github.com/php/php-src/pull/2560 :)
